From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Fri, 28 Apr 2023 23:43:57 +0530
Subject: android_webview: Fix build when webview_include_weblayer=false

gn gen errored with the below otherwise:

ERROR at //android_webview/system_webview_bundle.gni:12:23: Assignment had no effect.
  _base_target_name = get_label_info(invoker.base_module_target, "name")
                      ^------------------------------------------------
You set the variable "_base_target_name" here and it was unused before it went
out of scope.

Via these template invocations:
  system_webview_bundle("system_webview_bundle")  //android_webview/BUILD.gn:101

See //android_webview/BUILD.gn:101:3: whence it was called.
  system_webview_bundle("system_webview_bundle") {
  ^-----------------------------------------------
See //BUILD.gn:406:9: which caused the file to be included.
        "//android_webview:empty_group",
        ^------------------------------

Change-Id: If9a7b1f6da95d95a737a61e1b08f01fc26713d70
---
 android_webview/system_webview_bundle.gni | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/android_webview/system_webview_bundle.gni b/android_webview/system_webview_bundle.gni
--- a/android_webview/system_webview_bundle.gni
+++ b/android_webview/system_webview_bundle.gni
@@ -9,18 +9,21 @@ import("//weblayer/variables.gni")
 
 template("system_webview_bundle") {
   _is_trichrome = defined(invoker.is_trichrome) && invoker.is_trichrome
-  _base_target_name = get_label_info(invoker.base_module_target, "name")
-  _base_target_gen_dir =
-      get_label_info(invoker.base_module_target, "target_gen_dir")
-  _base_module_build_config =
-      "$_base_target_gen_dir/${_base_target_name}.build_config.json"
-  _rebased_base_module_build_config =
-      rebase_path(_base_module_build_config, root_build_dir)
-  _base_module_version_code =
-      "@FileArg($_rebased_base_module_build_config:deps_info:version_code)"
 
   assert(_is_trichrome == defined(invoker.static_library_provider))
 
+  if (webview_includes_weblayer) {
+    _base_target_name = get_label_info(invoker.base_module_target, "name")
+    _base_target_gen_dir =
+        get_label_info(invoker.base_module_target, "target_gen_dir")
+    _base_module_build_config =
+        "$_base_target_gen_dir/${_base_target_name}.build_config.json"
+    _base_module_version_code =
+        "@FileArg($_rebased_base_module_build_config:deps_info:version_code)"
+    _rebased_base_module_build_config =
+        rebase_path(_base_module_build_config, root_build_dir)
+  }
+
   if (webview_includes_weblayer && weblayer_in_split) {
     # TODO(crbug.com/1105096): If WebView starts using
     # //components/module_installer, it will probably make sense to refactor
-- 
2.40.1

